src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/AuditUtility.cs

```diff
@@ -86,6 +92,18 @@ public async Task CheckPackageVulnerabilitiesAsync(CancellationToken cancellatio
                 CheckPackageVulnerabilities(allVulnerabilityData.KnownVulnerabilities);
             }
 
+            bool HasPackages()
```

Comment:
Do you think this will have bad performance for very large restores?  It kind of looks like it will return very quickly as soon as any package reference is found and small projects that don't reference any packages will return quickly as well.  Just wondering your opinion in case my instincts are off.

Reply:
If you're asking if this new method will cause new perf problems, no, I don't think so. The restore graph is already resolved, so this method is just looking up a property on a collection of items in memory. And as you mentioned, it early exists when a package is found (vs only project references).

If the project doesn't have any packages, only project references, it's doing the same work that AuditUtility already does elsewhere (when it actually checks each package for known vulnerabilities). If the project doesn't have any packages, it'll be faster, since it won't need to obtain the vulnerability database. If the project does use at least one package, then it is additional work, but it early exits when it can, and even if it has to check a whole lot of Project references before it finds a package reference, it's just stuff in memory already.

My gut feeling is that it shouldn't have a big impact, but only a performance trace will tell us how often this method (or AuditUtility more generally) appears on sampled CPU stacks.